### ISSUE TITLE
Install Google fonts with Git 

### ISSUE THREAD

**Issue by lautis** [Homebrew/homebrew-cask#111460]
### Provide a detailed description of the proposed feature

Google fonts could be installed via Git instead of SVN. According to documentation, there isn't a Git strategy to download cask. Even more important for this case, we'd need an approach that only downloads relevant files.

Fetching only relevant files is possible with command line. For example, to get only source-serif-pro files:

```
git clone --depth 1 --filter=blob:none --no-checkout https://github.com/google/fonts
cd fonts
git checkout main -- ofl/sourceserifpro
```

The issue has been discussed before, e.g. https://github.com/Homebrew/homebrew-cask-fonts/issues/3970 and https://github.com/Homebrew/homebrew-cask/issues/80695. However, the resolution there was that SVN is the only available method to download only relevant files. Would it be possible to implement a Git strategy that fetches only relevant directory?

### What is the motivation for the feature?

Installing SVN (and it's dependencies) to be able to install packages is an extra step.

### Example use case

Google fonts are installed via SVN. For example, `brew install font-source-serif-pro`. Without SVN installed, this errors with `You must: brew install svn`.

**Comment by miccal** [comment]
Ping @vitorgalvao for your insight here, as you are most familiar with the `homebrew-cask-fonts` repository.

**Comment by vitorgalvao** [comment]
> Would it be possible to implement a Git strategy that fetches only relevant directory?

No, the limitation is in git itself. This has been discussed ad nauseam. We don’t want SVN, we use it because the alternative is considerably worse.

Installing SVN happens *once*.

This repo is too much trouble for what it’s worth, we don’t take feature requests for it.

**Comment by vitorgalvao** [comment]
> Would it be possible to implement a Git strategy that fetches only relevant directory?

No, the limitation is in git itself. This has been discussed ad nauseam. We don’t want SVN, we use it because the alternative is considerably worse.

If you want an alternative, you’ll have to come up with one which doesn’t download the whole repo for a single font *and be willing to implement it yourself*. This repo is more trouble than it’s worth and installing SVN happens *once*, it’s far from a problem.

**Comment by lautis** [comment]
@vitorgalvao to help me understand this better, is the shallow partial checkout approach in issue description insufficient in some way?

I understand that there isn’t motivation to change this given the straightforward workaround.

**Comment by vitorgalvao** [comment]
I misunderstood your solution, it seems. I was on a phone when I replied so couldn’t test it and relied on memory. That indeed looks acceptable. Reopening and pinging @reitermarkus.

**Comment by ran-dall** [comment]
> Would it be possible to implement a Git strategy that fetches only relevant directory?

@vitorgalvao or @reitermarkus Have we considered ever considereding using [sparse checkout](https://www.git-scm.com/docs/git-sparse-checkout)?

**Comment by vitorgalvao** [comment]
> THIS COMMAND IS EXPERIMENTAL. ITS BEHAVIOR, AND THE BEHAVIOR OF OTHER COMMANDS IN THE PRESENCE OF SPARSE-CHECKOUTS, WILL LIKELY CHANGE IN THE FUTURE.

Not a good idea to build a feature on it yet.

**Comment by ran-dall** [comment]
>> THIS COMMAND IS EXPERIMENTAL. ITS BEHAVIOR, AND THE BEHAVIOR OF OTHER COMMANDS IN THE PRESENCE OF SPARSE-CHECKOUTS, WILL LIKELY CHANGE IN THE FUTURE.

> Not a good idea to build a feature on it yet.

FWIW sparse checkout has been around for years (I see artcles about it from 2013, which must have been about more or less, when I started using it. So it must have been released before that.), and I haven't seen a breaking change to this functionality of git in some time. I think a year or two ago, they may have made some speed and stablity improvements, but nothing has changed to the point of breaking in years IME. NTM most glitches concerning sparse checkout has been either well-documented or resovled in that almost 10 years it's been around.

...but it really doesn't matter to me how you guys want to solve this. Just thought you should have all the facts, cause (as you well know) the documentation doesn't always accurately illustrates the whole picture, and often times, doesn't get updated as often as it probably should.

**Comment by vitorgalvao** [comment]
In that case, nothing else to add. I haven’t used the feature so I’ll defer.
